adaptive/learner/average_learner1D.py

```diff
@@ -0,0 +1,434 @@
+        return t_student * (variance_in_mean / n) ** 0.5
+
+    def tell_many(self, xs, ys):
+        """Tell the learner about some values.
+
+        Parameters
+        ----------
+        xs : Iterable of values from the function domain
+        ys : Iterable of lists of values (several data samples)
+             or scalars from the function image (mean values that
+             are directly included in self.data; the number of samples is then
+             assumed to be 1 and the error 0 for all data points; these points
+             will not be included in _rescaled_error_in_mean and therefore
+             will not be resampled)
+
+        Examples
+        ----------
+        If xs = [0,1] and ys = [[1.2,1.4],[2]], the y-values will be included in
+        self._data_samples.
+        If xs = [0,1] and ys = [1.2,2], the y-values will be directly included
+        in self.data, and the previous self.data[0] and self.data[1] will be erased.
+        """
+        # Check that all x are within the bounds
+        assert np.prod([x>=self.bounds[0] and x<=self.bounds[1] for x in xs]), 'x value out of bounds: remove x or enlarge the bounds of the learner'
+        # If data_samples is given:
+        if isinstance(ys[0], list):
+            for x, ys_ in zip(xs, ys):
+                y_avg = np.mean(ys_)
+                if x not in self.data:
+                    y = ys_.pop(0)
+                    self._update_data(x, y, "new")
+                    self._update_data_structures(x, y, "new")
+                if len(ys_):
+                    self.data[x] = y_avg
+                    self._data_samples.update({x: ys_+self._data_samples[x]})
+                    n = len(self._data_samples[x])
+                    self._number_samples[x] = n
+                    # self._update_data(x,y,"new") included the point
+                    # in _undersampled_points. We remove it if there are
+                    # more than min_samples samples, disregarding neighbor_sampling.
+                    if n > self.min_samples:
+                        self._undersampled_points.discard(x)
+                    self._error_in_mean[x] = self._calc_error_in_mean(self._data_samples[x], y_avg, n)
+                    self._update_distances(x)
+                    self._update_rescaled_error_in_mean(x, "resampled")
+                    if self._error_in_mean[x] <= self.min_Delta_g or n >= self.max_samples:
+                        self._rescaled_error_in_mean.pop(x, None)
+
+                    super()._update_scale(x, y_avg)
+                    self._update_losses_resampling(x, real=True)
+                    if self._scale[1] > self._recompute_losses_factor * self._oldscale[1]:
+                        for interval in reversed(self.losses):
+                            self._update_interpolated_loss_in_interval(*interval)
+                        self._oldscale = deepcopy(self._scale)
+
+        # If data is given:
+        else:
```

Comment:
@AlvaroGI, why is this functionality necessary BTW?

Reply:
This functionally allows to tell the learner about data that should be considered as "reliable", in the sense that it will not be resampled again. Theoretically, this data would correspond to the actual value of the function without noise, that is why there is no need to resample those points.
This functionality is not strictly necessary, but I was thinking in the following use-case scenario. Assume that you had some reliable previous data with low enough error bars that do not need resampling. If you want to add this data to the learner but you don't have access to all the samples anymore (only to the average value at each point), you can use this functionality, and the adaptive algorithm will take them into consideration when choosing a new point but will not resample them. I agree that this is not a common scenario but it may be useful for someone.

Reply:
I imagine that 99% of the people will exclusively use the `AverageLearner1D` with a `adaptive.Runner`. So in case this isn't really called, I vote for removing this option.

Maybe @akhmerov or @jbweston can chip in?

Reply:
In any case, using `tell_many` to indicate that a point shouldn't be sampled anymore isn't a good API. If that method desired, there should probably be another method doing that.

Reply:
I agree with @basnijholt: this feature changes what the function means, and we shouldn't do this.

We may separately consider if driving the learner like this is a relevant feature to already implement, but it certainly shouldn't be here.

Reply:
Makes sense! I removed this functionality in the [last implementation](https://github.com/python-adaptive/adaptive/pull/283/commits/a8c86ccae75e0f5d4da721bd1792f8c9a10c95e8) of `tell_many()` .